### PR TITLE
[VYMCA-110] Categories listing page + videos listing page

### PR DESCRIPTION
**Pull request by hamrant**
**Related Issue/Ticket:**

- https://3.basecamp.com/3608022/buckets/17045978/todos/3195270673
- https://fivejars.atlassian.net/browse/VYMCA-110
- https://fivejars.atlassian.net/browse/VYMCA-112

## Steps to test:

- [ ] Login to VY application
- [ ] Press on the categories link in the menu

screenshot-carnation openy docksal-2020 12 11-19_09_44

- [ ] Check that you can see all categories from all content types

Screenshot from 2020-12-11 19-12-41

- [ ] Click on the "Adjust" button and check that you can filter by content type and can change the sort order

Screenshot from 2020-12-11 19-12-51

- [ ] Open Any category from the list (For example Yoga)
- [ ] Check that you can see all content related to this category (empty content component should be hidden)

screenshot-carnation openy docksal-2020 12 16-15_21_59

- [ ]  Check that filters work fine and you can select content on this page or change sort order (Check that empty content component option disabled and can't be selected)

Screenshot from 2020-12-16 15-22-26

- [ ] Go to the home page and click "View All" on the videos
- [ ] Check that you can see all videos from the site

Screenshot from 2020-12-16 15-26-54

- [ ] Click on the "Adjust" button and check that you can filter by category and can change the sort order

Screenshot from 2020-12-16 15-28-35

- [ ] For more than 12 items will be enabled pager, you can change this on `/admin/openy/virtual-ymca` (Pager limit setting)
- [ ] Check that the "Videos" link from the main menu leads to this new page
- [ ] Repeat these steps for blogs
- [ ] Profit

**Note:** In this PR implemented logic part, styles will be fixed by @retif

### CODE REVIEW

**Review by retif — APPROVED** [review]

**Review by NightHunterSV — APPROVED** [review]

File: js/gated-content/src/router/index.js [hunk 1]
```diff
@@ -90,12 +104,24 @@ const routes = [
     meta: { requiresAuth: true },
   },
   {
-    path: '/categories',
-    redirect: '/categories/video',
+    path: '/blog-post',
+    redirect: { name: 'CategoryListing', query: { type: 'vy_blog_post' } },
```
Comment (NightHunterSV) [inline js/gated-content/src/router/index.js:108]: Probably should be redirected to BlogsListing?

Comment (hamrant) [reply js/gated-content/src/router/index.js:108]: yep, nice catch!

Comment (hamrant) [reply js/gated-content/src/router/index.js:108]: Fixed

**Comment by NightHunterSV** [comment]
image
Not sure if there is any sense to show the category if it "hasn't" enabled content. On the other side, there are virtual meetings in this category.

**Comment by NightHunterSV** [comment]
image
Is the "Content Type" correct label for the categories' filter?

File: js/gated-content/src/router/index.js [hunk 2]
```diff
@@ -90,12 +104,24 @@ const routes = [
...
   },
   {
-    path: '/blog-post',
-    redirect: '/categories/blog',
+    path: '/categories/blog',
+    redirect: { name: 'BlogsListing' },
```
Comment (NightHunterSV) [inline js/gated-content/src/router/index.js:112]: Probably should be exchanged with the above?

**Comment by NadyaRudenko** [comment]
retest

File: js/gated-content/src/router/index.js [hunk 3]
```diff
@@ -90,12 +104,24 @@ const routes = [
...
-    path: '/categories',
-    redirect: '/categories/video',
+    path: '/blog-post',
+    redirect: { name: 'CategoryListing', query: { type: 'vy_blog_post' } },
   },
   {
-    path: '/blog-post',
-    redirect: '/categories/blog',
+    path: '/categories/blog',
+    redirect: { name: 'CategoryListing', query: { type: 'vy_blog_post' } },
```
[the remark this answers is not in the document]
Comment (hamrant) [reply js/gated-content/src/router/index.js]: Fixed

File: js/gated-content/src/views/BlogsListingPage.vue [hunk 4]
```diff
@@ -0,0 +1,139 @@
+<template>
+  <div class="gated-content-videos-page">
+    <Modal v-if="showModal" @close="showModal = false">
+      <template v-slot:header>
+        <h3>Adjust</h3>
+      </template>
+      <template v-slot:body>
+        <div class="filter">
+          <h4>Content types</h4>
+          <div class="form-check" v-for="option in contentTypeOptions" v-bind:key="option.value">
```
[the remark this answers is not in the document]
Comment (hamrant) [reply js/gated-content/src/views/BlogsListingPage.vue]: Fixed

File: js/gated-content/src/views/BlogsListingPage.vue [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+              :value="option.value"
+              autocomplete="off"
+              v-model="preSelectedComponent"
+            >
+            <label :for="option.value">{{ option.label }}</label>
+          </div>
+        </div>
+        <div class="sort">
+          <h4>Sort order</h4>
+          <div class="form-check" v-for="option in filterOptions" v-bind:key="option.value">
```
[the remark this answers is not in the document]
Comment (hamrant) [reply js/gated-content/src/views/BlogsListingPage.vue]: Fixed

**Review by anpolimus — APPROVED** [review]

**Pull request merged**